### PR TITLE
Post a `Notification` when a `UserDefault` is set 

### PR DESCRIPTION
**Pull request by BenShutt**

### CODE REVIEW

File: Sources/UserDefault.swift [hunk 1]
```diff
@@ -49,6 +49,10 @@ public struct UserDefault<T> where T: UserDefaultElement {
                 } else {
                     UserDefaults.standard.set(object, forKey: key)
                 }
+
+                // Post on the `NotificationCenter`
+                NotificationCenter.default.post(userDefault: self, object: self)
```
Comment (BenShutt) [inline Sources/UserDefault.swift:54]: Note it's the `UserDefault<T>` which is posted, rather than the wrapped `T`.

Comment (simonmitchell) [reply Sources/UserDefault.swift:54]: What's the reasoning for this? :D My generics mind isn't in gear today :P

Comment (BenShutt) [reply Sources/UserDefault.swift:54]: Guess I thought it allows you to check both the `key` which it's saved under and the `wrappedValue` itself 😃

**Review by RodrigoLondon — APPROVED** [review]

File: Sources/UserDefault.swift (hunk 1 above)
Comment (simonmitchell) [reply Sources/UserDefault.swift:54]: Ahh yep of course! :D

**Review by simonmitchell — APPROVED** [review]

**Pull request merged**